reference/src/glossary.md

```diff
@@ -59,9 +59,22 @@ Moreover, the layout of a type records its *function call ABI* (or just *ABI* fo
 Note: Originally, *layout* and *representation* were treated as synonyms, and Rust language features like the `#[repr]` attribute reflect this. 
 In this document, *layout* and *representation* are not synonyms.
 
+#### Niche
+
+Invalid bit-patterns that will be used by layout optimizations.
```

Comment:
```suggestion
The *niche* of a type determines invalid bit-patterns that will be used by layout optimizations.
```

This is not a proper sentence. Also text should make sense if you skip the headers.